pkg/gui/gui.go

```diff
@@ -638,7 +642,7 @@ func (gui *Gui) runSubprocess(subprocess *exec.Cmd) error {
 	fmt.Fprintf(os.Stdout, "\n%s", style.FgGreen.Sprint(gui.Tr.PressEnterToReturn))
 	fmt.Scanln() // wait for enter press
 
-	return nil
+	return err
```

Comment:
This error will typically be something like 'exit status 1' which isn't very informative compared to the stuff that was written to stderr in the subprocess. What are your thoughts on something like this:
```go
func (gui *Gui) runSubprocess(subprocess *exec.Cmd) error {
	subprocess.Stdout = os.Stdout
	stderrBuf := &bytes.Buffer{}
	multiWriter := io.MultiWriter(os.Stdout, stderrBuf)
	subprocess.Stderr = multiWriter
	subprocess.Stdin = os.Stdin

	fmt.Fprintf(os.Stdout, "\n%s\n\n", style.FgBlue.Sprint("+ "+strings.Join(subprocess.Args, " ")))

	err := subprocess.Run()

	subprocess.Stdout = ioutil.Discard
	subprocess.Stderr = ioutil.Discard
	subprocess.Stdin = nil

	fmt.Fprintf(os.Stdout, "\n%s\n", style.FgGreen.Sprint(gui.Tr.PressEnterToReturn))
	fmt.Scanln() // wait for enter press

	if err != nil {
		return errors.New(fmt.Sprintf("%s\n%s", err, stderrBuf.String()))
	}

	return nil
}
```
You can test this out by opening up lazygit and pressing ':' then entering `>&2 echo "my error" && exit 1` (at least, this works for me on OSX)

Reply:
Hmm actually looks like using MultiWriter might cause problems as per https://stackoverflow.com/questions/33452726/golang-difference-between-os-stdout-and-multiwriter because isatty returns a different value depending on whether you're directly using os.Stdin. I can't actually find an example where I get different output though, but I wonder if that's the reason why I initially decided not to surface the error in lazygit (given `exit status 1` isn't super useful).

Off the top of my head here are our options:
1) continue returning no error (not great)
2) return an error for the sake of internal usage but don't actually surface it as a popup to the user given it's not super informative
3) use multiwriter and surface the error to the user including stderr
4) surface the 'exit status 1' error, without including anything from stderr

Each of these options appear to have issues. Option 2 is problematic when the user has disabled the 'press enter to return to lazygit' prompt because they would want to know if there was an error if they hadn't seen it in the output of the subprocess. Option 3 is problematic for the reasons described above. I'm currently leaning towards option 4. What do you think?

Reply:
I would also just go for option 4 right now.

At least I can still see the error message, as whenever a script like this is run (e.g. by a pre-commit hook, or with `:`), the view changes back to the terminal, and I see the output including the complete error message together with a "Press enter to return to lazygit" text. Afterwards I only see "Error exit status 1" in a pop-up box in lazygit itself, but I already know what the problems at that time, so I don't really mind the non-descriptive text.

Not sure though, if this is just my configuration, or if there are some options that hide the initial output.

Reply:
Yeah there's soon to be an option to jump straight back to lazygit without the 'press enter to return' part, which means for those who enable that option, they might be missing the actual error text. But I'm happy to still go with option 4 for now just because it has the fewest problems